pkg/util/wsstream/conn.go

```diff
@@ -175,20 +209,44 @@ func (conn *Conn) Open(w http.ResponseWriter, req *http.Request) ([]io.ReadWrite
 	return rwc, nil
 }
 
+func splitProtocol(proto string, b64, raw string) (string, string) {
+	if proto == b64 || proto == raw {
+		return "", proto
+	}
+	if strings.HasSuffix(proto, "."+b64) {
```

Comment:
Don't do this - instead, accept a map of protocol -> behavior structs, and base your decisions on structured fields in the struct.  I.e.

```
type map[string]Behavior
type Behavior struct {
  Binary bool
  ChannelHandlers [](functions that handle channels)
}